iosApp/iosApp/ContentView.swift

```diff
@@ -286,16 +286,18 @@ struct ContentView: View {
                             viewportProvider: viewportProvider
                         )
                         .toolbar(.hidden, for: .tabBar)
-                        .onAppear {
-                            let filtered = stopFilter != nil
-                            screenTracker.track(
-                                screen: filtered ? .stopDetailsFiltered : .stopDetailsUnfiltered
-                            )
-                        }
                     }
                     // Set id per stop so that transitioning from one stop to another is handled by removing
                     // the existing stop view & creating a new one
                     .id(stopId)
+                    .onChange(of: stopId) { nextStopId in stopDetailsVM.handleStopChange(nextStopId) }
+                    .onAppear {
+                        stopDetailsVM.handleStopAppear(stopId)
+                        screenTracker.track(
+                            screen: stopFilter != nil ? .stopDetailsFiltered : .stopDetailsUnfiltered
```

Comment:
Sorry @EmmaSimon one last thought - does this affect how visits to stop details are counted in GA? Wondering if it would alter the stats about how users progress through the funnel

Reply:
I think it might actually, I should move it back to where it was and add a second `onAppear` here.